src/secure_mount.rs

```diff
@@ -30,100 +34,126 @@ fn check_mount(secure_dir: &str) -> bool {
         }
 
         if tokens[2] == secure_dir {
-            if &tokens[0] != &"tmpfs" {
-                error!("secure storage location {} already mounted on wrong file system type: {}.  Unmount to continue.", secure_dir, tokens[0]);
-                debug!(
-                    "secure storage location {} aleady mounted on tmpfs",
-                    secure_dir
-                );
+            if tokens[0] != "tmpfs" {
+                return emsg(format!("secure storage location {} already mounted on wrong file system type: {}.  Unmount to continue.", secure_dir, tokens[0]).as_str(), None::<String>);
             } else {
-                return true;
+                return Ok(true);
             }
         }
     }
-    debug!("secure storage location {} not mounted.", secure_dir);
-    false
+
+    info!("secure storage location {} not mounted.", secure_dir);
+    Ok(false)
 }
 
 /*
- * Return: Result contains secure mount directory or error code
+ * Return: Result wrap secure mount directory or error code
  *
  * Mounted the work directory as tmpfs, which is owned by root. Same
  * implementation as the original python version, but the chown/geteuid
  * functions are unsafe function in Rust to use.
  */
-fn mount() -> Result<String, i32> {
-    // use /tmpfs-dev directory if MOUNT_SECURE flag is setm, which doesn't
-    // mount to the system. This is for developement envrionment.
+fn mount() -> Result<String, Box<String>> {
+    // Use /tmpfs-dev directory if MOUNT_SECURE flag is setm, which doesn't
+    // mount to the system. This is for developement envrionment. No need to
+    // mount to file system.
     if !common::MOUNT_SECURE {
         let secure_dir = format!("{}{}", common::WORK_DIR, "/tmpfs-dev");
         let secure_dir_path = Path::new(secure_dir.as_str());
         if !secure_dir_path.exists() {
             match fs::create_dir(secure_dir_path) {
                 Ok(()) => {
-                    return Ok(secure_dir_path.to_str().unwrap().to_string())
+                    info!("Directory {:?} created.", secure_dir_path);
                 }
                 Err(e) => {
-                    error!("Failed to create directory, error {}", e);
-                    return Err(-1);
+                    return emsg("Failed to create directory.", Some(e));
                 }
             }
+            if let Some(s) = secure_dir_path.to_str() {
+                return Ok(s.to_string());
+            } else {
+                return emsg("Failed to get the path string.", None::<String>);
+            }
         }
     }
 
+    // Monut the directory to file system
     let secure_dir = format!("{}{}", common::WORK_DIR, "/secure");
+    match check_mount(&secure_dir) {
+        Ok(b) => {
```

Comment:
Match on `Ok(false)` and `Ok(true)` rather than having an `if` inside the `match`.

Reply:
Changed.